libraries/common/stream.effekt

```diff
@@ -409,6 +409,86 @@ def each(string: String): Unit / emit[Char] =
 def collectString { stream: () => Unit / emit[Char] }: String =
   returning::collectString[Unit]{stream}.second
 
+effect snapshot(): Unit
```

Comment:
Nitpick: could we hide `snapshot` in some `internal` [`teeing` ?] namespace so that it doesn't look like a part of the API?

Reply:
Done.